paddle/framework/operator.cc

```diff
@@ -186,6 +186,47 @@ void OperatorBase::GenerateTemporaryNames() {
   }
 }
 
+template <>
+const Tensor* InferShapeContext::Input<Tensor>(const std::string& name) const {
+  auto* var = InputVar(name);
+  return var == nullptr ? nullptr : GetTensorFromVar(var);
+}
+
+template <>
+const std::vector<const Tensor*> InferShapeContext::MultiInput<Tensor>(
+    const std::string& name) const {
```

Comment:
Specialized template function `InferShapeContext::Input` and `InferShapeContext::MultiInput`, thus, the `InferShape` of non-sequence operators can call `Input<Tensor>` to avoid to use `LoDTensor`.

Reply:
I think this specialization is great to hide the difference between LoDTensor and Tensor. I can also accept that we don't have this and expose LoDTensor to non-sequence operator contributors, because I feel that LoDTensor is a key difference from TensorFlow and could be a PaddlePaddle flag.